### DISCUSSION TITLE
GitHub API download from private repo stops at 205K

### DISCUSSION THREAD

**Discussion by sjafroudi** [community/community#36175] [Apps, API and Webhooks]
Hello,

I'm having issues using the GitHub API to download files from a private repo. There are 4 files, one of them considerably larger than the others, but still not over the 100MB download limit:

- file.dbf (1.77 MB)
- file.prj (165 Bytes)
- file.shp (75 MB)
- file.shx (75.5 KB)

Currently have a method for downloading the file programmatically, which downloads everything except for the `.shp` file. I try to use cURL to download the missing file, 

`curl -H "Accept: application/vnd.github+json" -H "Authorization: token <token>" https://github.com/<organization>/path/to/file/file.shp --output /tmp/github/<organization>/path/to/file/file.shp`

but nothing seems to work consistently (also have tried using `Bearer` in place of `token` in the above snippet). This is the output when I try to use cURL to download the missing `.shp` file: 

```
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  205k    0  205k    0     0   170k      0 --:--:--  0:00:01 --:--:--  171k
```

Only 205KB are downloaded but the file itself is 75MB. I've been having this problem for the past week or so and sometimes it works, other times it doesn't. Any insights would be appreciated, thanks!

**Comment by FahdSeddik** [comment]
Hey @sjafroudi 
There doesn't seem to be a problem. Make sure to have a stable connection. Hope this helps!

**Reply by sjafroudi** [reply]
Thank you @FahdSeddik. I don't think internet connection is the issue here, however.

**Comment by jfullstackdev** [comment]
it might be GitHub LFS ? 

check this out 

https://stackoverflow.com/questions/58219778/how-to-get-a-direct-link-to-a-file-in-github-lfs-large-file-storage

**Reply by sjafroudi** [reply]
Thank you @jdevstatic! In this case, I don't believe that's the issue. Git LFS is used when you're trying to push files over 100MB to a remote repo. My issue is that I need to download a 75MB file from a private remote repo.
